server/components/populate_league_test.js

```diff
@@ -10,7 +11,7 @@ var Populate = (function() {
     var _fillLeague = function(user) {
         if (user.role !== 'admin') {
             console.log("Error, user is not an admin, user role is: " + user.role);
-            return false;
+            BPromise.reject("User is not an admin");
```

Comment:
It's not sufficient just to call BPromise.reject() -- you want to 
return BPromise.reject("User is not an admin");